tests/client/amo/components/TestAddonCompatibility.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import {
+  renderIntoDocument,
+  findRenderedComponentWithType,
+} from 'react-addons-test-utils';
+import { findDOMNode } from 'react-dom';
+import { Provider } from 'react-redux';
+
+import AddonCompatibility from 'amo/components/AddonCompatibility';
+import createStore from 'amo/store';
+import { fakeAddon, signedInApiState } from 'tests/client/amo/helpers';
+import { getFakeI18nInst, userAgents } from 'tests/client/helpers';
+import I18nProvider from 'core/i18n/Provider';
+
+
+describe('AddonCompatibility', () => {
+  function render({ ...props }) {
+    return findRenderedComponentWithType(renderIntoDocument(
+      <Provider store={createStore({ api: signedInApiState })}>
+        <I18nProvider i18n={getFakeI18nInst()}>
+          <AddonCompatibility addon={fakeAddon} {...props} />
+        </I18nProvider>
+      </Provider>
+    ), AddonCompatibility);
+  }
+
+  it('renders nothing if add-on is compatible', () => {
+    const root = render({ isCompatibleWithUserAgent: () => true });
+    const rootNode = findDOMNode(root);
+
+    assert.equal(rootNode, null);
+  });
+
+  it('does not show a notice for a version at the max version', () => {
+    const root = render({
+      maxVersion: '10.0',
+      // This is Firefox 10 on Linux.
+      userAgent: userAgents.firefox[0],
+    });
+    const rootNode = findDOMNode(root);
+
+    assert.equal(rootNode, null);
+  });
+
+  it('renders component if add-on is not compatible', () => {
+    const root = render({ isCompatibleWithUserAgent: () => false });
+    const rootNode = findDOMNode(root);
+
+    assert(rootNode);
+  });
+
+  it('renders a notice for non-Firefox browsers', () => {
+    const root = render({
+      isCompatibleWithUserAgent: () => false,
+      userAgent: userAgents.chrome[0],
+    });
+    const rootNode = findDOMNode(root);
+
+    // This is localised to the current locale, which is en-US in our
+    // signedInApiState helper.
+    assert.equal(rootNode.querySelector('a').href,
+      'https://www.mozilla.org/en-US/firefox/');
```

Comment:
to avoid mistakes, I prefer to see these kind of tests done with a non-default lang. You could alter `render()` to accept an override to the `signedInApiState` or something like that.